### PR TITLE
feat(repository): allow custom keyFrom for hasMany and hasOne

### PR DESCRIPTION
**Pull request by agnes512**
## Description

LB4 relation `hasMany` and `hasOne` use the mode id property as the source key. However, there are cases use would like to define a non-id property as their source key in these two relations ( in the other words, join tables on non-id attribute). 

### Example

- `Customer` has many `Order`s. 
- `Customer` has many `Review`s.

Screen Shot 2019-12-16 at 3 40 45 PM

### Current Behavior

Traditionally, `Order` takes the id property `Customer.customerId` as it's foreign key.

### Expect Behavior

**This feature would allow `Review` to have the non-id property `Customer.articleId` as its foreign key.**

Users could travers order and also reviews with different attributes.

## Implementation

Similar to the `belongTo` relation, this PR enables the custom source key, which is `keyFrom` in lb4 relation metadata. The feature should also work with inclusion.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/HasMany-relation.md [hunk 1]
```diff
@@ -265,6 +265,54 @@ export class Order extends Entity {
 }
 ```
 
+If you need to use another attribute other than the id property to be the source
+key, customizing `keyFrom` filed would allow you to do so:
```
Comment (nabdelgadir) [inline docs/site/HasMany-relation.md]: typo: `filed` -> `field`

File: docs/site/hasOne-relation.md [hunk 2]
```diff
@@ -201,6 +201,54 @@ export class Account extends Entity {
 }
 ```
 
+If you need to use another attribute other than the id property to be the source
+key, customizing `keyFrom` filed would allow you to do so:
```
Comment (nabdelgadir) [inline docs/site/hasOne-relation.md]: similar typo: `filed` -> `field`

File: packages/repository-tests/src/crud/relations/fixtures/models/shipment.model.ts [hunk 3]
```diff
@@ -22,10 +22,16 @@ export class Shipment extends Entity {
...
+    type: 'number',
+  })
+  shipmentId: number;
+
   @property({type: 'string'})
   name: string;
 
-  @hasMany(() => Order, {keyTo: 'shipment_id'})
+  // keyFrom is not the id property of Shipment
+  @hasMany(() => Order, {keyFrom: 'shipmentId', keyTo: 'shipmentInfo'})
```
Comment (nabdelgadir) [inline packages/repository-tests/src/crud/relations/fixtures/models/shipment.model.ts]: The point of `shipment_id` was to test it with snake case. Is there somewhere else that tests it?

Comment (agnes512) [reply packages/repository-tests/src/crud/relations/fixtures/models/shipment.model.ts]: reversed

File: packages/repository-tests/src/crud/relations/acceptance/has-many-inclusion-resolver.relation.acceptance.ts [hunk 4]
```diff
@@ -90,17 +93,16 @@ export function hasManyInclusionResolverAcceptance(
...
             },
           ],
         }),
       ]);
     });
 
     it('returns multiple model instances including related instances', async () => {
-      const thor = await customerRepo.create({name: 'Thor'});
-      const odin = await customerRepo.create({name: 'Odin'});
+      const thor = await customerRepo.create({ name: 'Thor' });
```
Comment (nabdelgadir) [inline packages/repository-tests/src/crud/relations/acceptance/has-many-inclusion-resolver.relation.acceptance.ts]: unrelated whitespace changes?

File: docs/site/hasOne-relation.md [hunk 5]
```diff
@@ -201,6 +201,54 @@ export class Account extends Entity {
...
+export class Foo extends Entity {
+  // constructor, properties, etc.
+
+  // specify the keyTo if the source key is not the id property
+  @belongsTo(() => Supplier, {keyTo: 'anotherKey'})
+  supplierId: number; // default foreign key name
+}
+```
+
+{% include important.html content="LB4 doesn't support composite keys for now. e.g join two tables with more than one source key. Related GitHub issues: [Composite primary/foreign keys](https://github.com/strongloop/loopback-next/issues/1830)" %}
```
Comment (nabdelgadir) [inline docs/site/hasOne-relation.md]: nitpick: `issues` -> `issue`
`e.g join` -> `e.g. joining`

File: docs/site/hasOne-relation.md [hunk 6]
```diff
@@ -201,6 +201,54 @@ export class Account extends Entity {
...
+    id: true,
+  })
+  id: number;
+
+  // if you'd like to use this property as the source id
+  // of a certain relation that relates to a model `Foo`
+  @property({
+    type: 'number',
+  })
+  anotherKey: number;
```
Comment (jannyHou) [inline docs/site/hasOne-relation.md]: same here :)

File: docs/site/HasMany-relation.md [hunk 7]
```diff
@@ -265,6 +265,54 @@ export class Order extends Entity {
...
+  id: number;
+
+  // if you'd like to use this property as the source id
+  // of a certain relation that relates to a model `Foo`
+  @property({
+    type: 'number',
+  })
+  anotherKey: number;
+
+  @hasMany(() => Foo, {keyFrom: 'anotherKey'})
```
Comment (jannyHou) [inline docs/site/HasMany-relation.md]: nitpick: maybe use a more valid example here instead of `Foo`? Like what you have in the tests.
